src/rest_vol_dataset.c

```diff
@@ -3744,3 +3764,155 @@ dataset_read_scatter_op(const void **src_buf, size_t *src_buf_bytes_used, void *
+    if (npoints < 2)
+        FUNC_GOTO_DONE(true);
+
+    if ((ndims = H5Sget_simple_extent_ndims(space_id)) < 0)
+        FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't retrieve dataspace dimensionality");
+    if (!ndims)
+        FUNC_GOTO_DONE(true);
+
+    switch (H5Sget_select_type(space_id)) {
+        case H5S_SEL_HYPERSLABS:
+            if (NULL == (dims = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*dims))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for dimension 'dims' values");
+
+            if (H5Sget_simple_extent_dims(space_id, dims, NULL) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't get dataspace dimension size");
+
+            if (NULL == (start = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*start))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'start' values");
+            if (NULL == (stride = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*stride))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'stride' values");
+            if (NULL == (count = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*count))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'count' values");
+            if (NULL == (block = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*block))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, FAIL,
+                                "can't allocate space for hyperslab selection 'block' values");
+
+            if (nblocks = H5Sget_select_hyper_nblocks(space_id) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't get number of hyperslab blocks");
+
+            if (H5Sget_regular_hyperslab(space_id, start, stride, count, block) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, FAIL, "can't get regular hyperslab selection");
+
+            /* For contiguous, the stride should be 1. */
+            for (i = 0; i < ndims; i++) {
+                if (stride[i] > 1)
+                    FUNC_GOTO_DONE(false);
+            }
+
+            if (nblocks > 1) {
+                /* Multiple blocks: count should be 1 except for the last dimension (fastest) */
+                for (i = 0; i < ndims - 1; i++) {
+                    if (count[i] > 1)
+                        FUNC_GOTO_DONE(false);
+                }
+            }
+
+            /* For contiguous, all faster running dimensions than the current dimension should be selected
+             * completely */
+            whole = (start[ndims - 1] == 0) && (count[ndims - 1] * block[ndims - 1] == dims[ndims - 1]);
```

Comment:
This seems to require that the fastest running/last dimension be selected completely, which I don't think is necessary for the selection to be contiguous.

Should the comment say that all _slower_ running dimensions than the last dimension must be contiguous?

Reply:
I had the same thought yesterday until I did a small test which shows it was a wrong thought. Now I read contiguous selection as a selection which is contiguous in memory. If the fastest running/last dimension is not completely selected you need additional jumps to get the right location in memory. A way to visual this, at least for me, is by considering a 2D example and make various selections. For each selection translate it to which memory locations you need to get the selected data.